lib/azure/AzureDataStore.ts

```diff
@@ -70,52 +71,40 @@ export default class AzureDataStore {
    * @param keys
    * @param callback
    */
-  private listKeys (prefix: string, currentToken: any, keys: any, callback: any): void {
-    if (typeof callback === 'undefined') {
-      callback = keys;
-      keys = [];
-    }
-
-    this.blobService.listBlobsSegmentedWithPrefix(this.container, prefix, currentToken, (err, result, response) => {
-      if (err) {
-        return callback(new Error(err.name));
-      }
-      if (response.isSuccessful) {
-        result.entries.forEach((d) => {
-          keys.push(new Key(d.name.slice(this.path.length), false));
-        });
-
-        if (result.continuationToken) {
-          return this.listKeys(prefix, result.continuationToken, keys, callback);
+  private async listKeys (prefix: string, currentToken: any, keys: any): Promise<any> {
+    return new Promise((resolve, reject) => {
+      this.blobService.listBlobsSegmentedWithPrefix(this.container, prefix, currentToken, async (err, result, response) => {
+        if (err) {
+          reject(new Error(err.name));
+          return;
+        }
+        if (response.isSuccessful) {
+          result.entries.forEach((d) => {
+            const testing = d.name.slice(this.path.length);
+            keys.push(new Key(d.name.slice(this.path.length), false));
+          });
+
+          if (result.continuationToken) {
+            await this.listKeys(prefix, result.continuationToken, keys);
+            resolve(keys);
+            return;
+          }
         }
-      }
 
-      callback(err, keys);
+        resolve(keys);
+      });
     });
   }
 
   /**
-   * Returns an iterator for fetching objects from azure by their key
-   * @param keys
-   * @param keysOnly Whether or not only keys should be returned
+   * returns an asyncIterator that yields all the keys and values in the store
    */
-  private getBlobIterator (keys: any, keysOnly: boolean): any {
-    let count: number = 0;
-
-    return {
-      next: (callback: any) => {
-        if (count >= keys.length) {
-          return callback(null, null, null);
-        }
-        let currentKey = keys[count++];
-        if (keysOnly) {
-          return callback(null, currentKey, null);
-        }
-        this.get(currentKey, (err: any, data: any) => {
-          callback(err, currentKey, data);
-        });
-      }
-    };
+  private async * _all () {
```

Comment:
This is a private function to be implemented and used by the default implementation of query